bundles/core/src/main/java/com/adobe/cq/commerce/demandware/preview/PreviewServiceImpl.java

```diff
@@ -53,104 +56,75 @@
-import com.day.cq.wcm.api.Page;
-import com.day.cq.wcm.api.PageManager;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
 
 /**
  * Component previewComponent service to render component previewComponent for Demandware placeholder components. The rendered previewComponent
  * will be cached by a simple cache, avoid unnecessary backend request.
  */
-@Component(metatype = true, policy = ConfigurationPolicy.REQUIRE, label = "Demandware Component Preview Service")
+@Component(metatype = false, policy = ConfigurationPolicy.REQUIRE, label = "Demandware Component Preview Service")
 @Service()
 public class PreviewServiceImpl implements PreviewService {
-
+    
     private static final Logger LOG = LoggerFactory.getLogger(PreviewServiceImpl.class);
     private static final String ERROR_MSG = "<p>Live Demandware preview failed to render. See log for details.</p>";
-
-    @Property(label = "Preview endpoint")
-    private static final String PREVIEW_PAGE_ENDPOINT = "endpointPage";
-
-    @Property(label = "Preview endpoint for category pages")
-    private static final String PREVIEW_SEARCH_ENDPOINT = "endpointSearch";
-
-    @Property(label = "Preview template path")
-    private static final String PREVIEW_TEMPLATE = "template";
-
-    @Property(label = "Preview default site")
-    private static final String PREVIEW_DEFAULT_SITE = "site";
-
-    @Property(label = "Enable preview cache", boolValue = true)
-    private static final String PREVIEW_CACHE_ENABLED = "cache.enabled";
-
-    private static final int DEFAULT_PREVIEW_CACHE_TIME = 60;
-    @Property(label = "Caching time in seconds", intValue = DEFAULT_PREVIEW_CACHE_TIME)
-    private static final String PREVIEW_CACHE_TIME = "cache.time";
-
-    @Property(label = "Enable storefront protection", boolValue = false)
-    private static final String STORFRONT_PROTECTION_ENABLED = "storefront.protected";
-
-    @Property(label = "Protected storefront user")
-    private static final String STORFRONT_PROTECTION_USER = "storefront.user";
-
-    @Property(label = "Protected storefront password")
-    private static final String STORFRONT_PROTECTION_PASSWORD = "storefront.password";
-
-
+    
     @Reference
     DemandwareClientProvider clientProvider;
-
+    
     @Reference
     RenderService renderService;
-
-    private String previewPageEndPoint;
-    private String previewSearchEndPoint;
-    private String previewTemplate;
-    private String previewDefaultSite;
-
-    private PreviewCache cache;
-
+    
+    @Reference
+    private InstanceIdProvider instanceId;
+    
+    @Reference
+    private PreviewServiceConfigProvider previewServiceConfigProvider;
+    
     private CredentialsProvider credentialsProvider;
-
-
+    
+    
     @Override
     public String previewComponent(Resource resource, boolean useCache) {
+        PreviewServiceConfig previewServiceConfig = getPreviewServiceConfig(resource);
```

Comment:
1) getPreviewServiceConfig(resource) can cause nullpointerexception because it is called before if nullcheck (resource != null)
2) there is no need to get previewServiceConfig in previewComponent methotd. You can get it instead inside of getPreviewContent method and thus decrease parameters for this method:

`private String getPreviewContent(final Resource resource, final boolean useCache, final List<NameValuePair> params, 
                                     final String... selectores) {
        String renderedPreview = "";
        PreviewServiceConfig previewServiceConfig = getPreviewServiceConfig(resource);
        final String endPoint = previewServiceConfig.getPreviewPageEndPoint();`